lib/src/repo.rs

```diff
@@ -1664,33 +1665,38 @@ impl MutableRepo {
     }
 
     fn merge_view(&mut self, base: &View, other: &View) -> BackendResult<()> {
-        // Merge working-copy commits. If there's a conflict, we keep the self side.
-        for (workspace_id, base_wc_commit) in base.wc_commit_ids() {
-            let self_wc_commit = self.view().get_wc_commit_id(workspace_id);
-            let other_wc_commit = other.get_wc_commit_id(workspace_id);
-            if other_wc_commit == Some(base_wc_commit) || other_wc_commit == self_wc_commit {
-                // The other side didn't change or both sides changed in the
-                // same way.
-            } else if let Some(other_wc_commit) = other_wc_commit {
-                if self_wc_commit == Some(base_wc_commit) {
-                    self.view_mut()
-                        .set_wc_commit(workspace_id.clone(), other_wc_commit.clone());
-                }
+        let mut workspace_updates = vec![];
+        for (workspace_id, (my_commit, other_commit)) in
+            diff_named_commit_ids(self.view().wc_commit_ids(), other.wc_commit_ids())
```

Comment:
We compare base and other to get around borrowing issue. I think we can do the same:
```rust
        let changed_wc_commits = diff_named_commit_ids(base.wc_commit_ids(), other.wc_commit_ids());
        for (workspace_id, (base_target, other_target)) in changed_wc_commits {
            let self_target = self.view().get_wc_commit_id(workspace_id);
            if let Some(&new_target) = trivial_merge(&[self_target, base_target, other_target]) {
                if let Some(target) = new_target.cloned() {
                    self.view_mut().set_wc_commit(workspace_id.clone(), target);
                } else {
                    self.view_mut().remove_wc_commit(workspace_id);
                }
            } else // special case for merge conflict
```

Reply:
I think I'll extract this to a separate PR.

It seems to me this might not have the same (strange IMO; I'm guessing it's a workaround around... something) behavior if `other_target` is `None`.

Even if it does, I want to try writing it out in `if` statements, I think that might look clearer.

Reply:
Moved to https://github.com/jj-vcs/jj/pull/5993.

Reply:
I think the intent here is that deletion is always prioritized, no matter if the workspace was deleted at the `self` side or the `other` side.

Reply:
Yes, but why does this make sense?

Reply:
I assume the expectation is that `jj workspace forget` is stronger than e.g. `jj new` because the workspace is forgotten anyway. I don't know if that's always true, but it would be more surprising if `jj workspace forget` was cancelled?

Reply:
Yeah, I'm not sure. Let's continue this in <https://github.com/jj-vcs/jj/pull/5993#discussion_r1990511541>, though my reply there doesn't say much.